docs/classic-ui/theming/diazo.md

```diff
@@ -0,0 +1,136 @@
+
+```console
+.
+├── index.html
+├── manifest.cfg
+├── package.json
+├── README.rst
+├── rules.xml
+├── styles
+│   ├── theme.min.css
+│   └── theme.scss
+└── tinymce-templates
+    ├── bs-dark-hero.html
+    ├── bs-hero-left.html
+    └── bs-pricing.html
+```
+
+
+## Integrate an external theme using Diazo
+
+Begin with any theme from outside Plone.
+It should be composed of the required HTML, CSS, and JavaScript files.
+
+CSS and JavaScript files should be properly versioned or hashed to avoid any caching problems whenever the theme is updated.
+
+If the CSS file is called {file}`global.css`, and your designer updates the CSS without changing the file name, you will surely face caching issues.
+Browsers, varnish, or other proxy servers might cache your files, and not serve them to the end users until the cache expires or gets flushed.
+
+To avoid this issue, CSS bundling techniques that use npm tooling—such as Gulp, Grunt, or Webpack—create hashed or versioned filenames for CSS and JavaScript files.
+The following HTML snippets show examples of versioned files.
+
+```html
+  <link id="frontend-css" rel="stylesheet" href="./css/app.css?v=14" />
+  <script id="frontend-javascript" src="./js/app.js?v=3"></script>
+```
+
+When you have all files of your theme, put them in the {file}`theme` folder, and organize the CSS and JavaScript folders as you received them from your designer.
+
+You may want to remove the `plonecli` generated `styles` and `tinymce-templates` folders.
+
+
+## Adjust the theme manifest
+
+Open the {file}`manifest.cfg` file.
+You will see the following lines.
+
+```cfg
+production-css = ++theme++my-shiny-theme/styles/theme.min.css
+tinymce-content-css = ++theme++my-shiny-theme/styles/theme.min.css
+```
+
+Comment them out as shown.
+
+```cfg
+# production-css = ++theme++my-shiny-theme/styles/theme.min.css
+# tinymce-content-css = ++theme++my-shiny-theme/styles/theme.min.css
+```
+
+This way, you signal that you don't want either Plone or Diazo to manage the CSS files at all.
+But it means that you will need to handle them in your design HTML files.
+
+
+## HTML template structure
+
+Your theme should be as simple as possible.
+That will make your {file}`rules.xml` file also as simple as possible.
+
+The {file}`rules.xml` file declares which parts of theme will be replaced by the HTML produced by Plone.
+
+It is a good practice to have a `<div>` element called `content` in your theme, which will contain the maximum space of the content area of your site.
+That way you can inject the HTML produced by Plone there using Plone's content section too.
+
+Add a stanza in your {file}`rules.xml` file.
+
+```xml
+<replace
+        css:theme-children="#content"
+        css:content-children="article#content"
+        css:if-content="#content"/>
+```
+
+
+## How to theme using Diazo
+
+You can start with the provided {file}`rules.xml` file.
+You will need to write your own rules to bring the dynamic content from Plone into the theme.
```

Comment:
How will I write any rules, when there is no reference documentation linked here?

Reply:
The reference of writing those rules is at https://docs.diazo.org/en/latest/basic.html